NeuroIDAdvancedDevice/AdvancedDevice.swift

```diff
@@ -12,22 +12,26 @@ struct NIDResponse: Codable {
     let key: String
 }
 
-public class NeuroIDADV: NSObject {
-    public static func getAdvancedDeviceSignal(
-        _ apiKey: String,
-        completion: @escaping (Result<String, Error>) -> Void
-    ) {
-        getAPIKey(apiKey) { result in
-            switch result {
-            case .success(let fAPiKey):
-                retryAPICall(apiKey: fAPiKey, maxRetries: 3, delay: 2) { result in
-                    completion(result)
-                }
-            case .failure(let error):
-                completion(.failure(error))
-            }
-        }
-    }
+/** Interface that allows for testing */
+protocol DeviceSignalService {
+    func getAdvancedDeviceSignal(_ apiKey: String, completion: @escaping (Result<String, Error>) -> Void)
+}
+
+public class NeuroIDADV: NSObject, DeviceSignalService {
+    public static let shared = NeuroIDADV()
+    
+    public func getAdvancedDeviceSignal(_ apiKey: String, completion: @escaping (Result<String, Error>) -> Void) {
+           NeuroIDADV.getAPIKey(apiKey) { result in
```

Comment:
Is there a reason we specify NeuroIDADV here?

Reply:
Yes, this indicates we are accessing it statically instead of via our singleton instance.